cmd/catalyst-node/catalyst-node.go

```diff
@@ -434,25 +444,47 @@ func redirectHlsHandler() http.Handler {
 		lat := r.Header.Get("X-Latitude")
 		lon := r.Header.Get("X-Longitude")
 
-		nodeAddr, err := getClosestNode(playbackID, lat, lon)
-		if err != nil {
+		var nodeAddr, validPrefix string
+		var err error
+
+		for _, prefix := range redirectPrefixes {
+			nodeAddr, err = getClosestNode(playbackID, lat, lon, prefix)
+			if err != nil {
+				glog.Errorf("error finding origin server playbackID=%s prefix=%s error=%s", playbackID, prefix, err)
+				continue
+			}
+			if nodeAddr != "" {
+				validPrefix = prefix + "+"
+				err = nil
+				break
+			}
+		}
+
+		if nodeAddr == "" || err != nil {
 			glog.Errorf("error finding origin server playbackID=%s error=%s", playbackID, err)
 			w.WriteHeader(http.StatusNotFound)
 			return
 		}
 
-		rURL := fmt.Sprintf("%s://%s/hls/%s/index.m3u8", protocol(r), nodeAddr, playbackID)
+		rURL := fmt.Sprintf("%s://%s/hls/%s%s/index.m3u8", protocol(r), nodeAddr, validPrefix, playbackID)
+		glog.V(6).Infof("generated redirect url=%s", rURL)
 		http.Redirect(w, r, rURL, http.StatusFound)
 	})
 }
 
 func parsePlaybackID(path string) (string, bool) {
-	r := regexp.MustCompile("^/hls/([a-zA-Z0-9_\\-+]+)/index.m3u8$")
+	r := regexp.MustCompile("^/hls/([\\w+-]+)/index.m3u8$")
```

Comment:
Good change! 👏 

One thing I wonder is if it shouldn't be `\\w\\-+`? Don't we need to include `\`?

Reply:
We aren't including `\`; it was just to escape the `-` character in original pattern. Otherwise it'd be interpreted as character set from `_` to `+`; moving `-` to beginning or end of character set doesnt need to escape it.

Reply:
Are underscores still included here? They should be.

Reply:
yes, `\w` comes from perl; consisting of `[a-zA-Z0-9_]`. referring from docs at: https://golang.org/s/re2syntax